src/encoded/static/components/encyclopedia.js

```diff
@@ -225,29 +126,29 @@ const Encyclopedia = (props, context) => {
         setdownloadHref(newHref);
     };
 
-    // Update assembly, organism, browser files, and download link when user clicks on tab
     const handleTabClick = (tab) => {
-        setVizFiles([]);
-        setSelectedOrganism(tab);
-        setAssembly(defaultAssemblyByOrganism[tab]);
-
-        // Update download link
-        const query = new QueryString(downloadHref);
+        const query = new QueryString(props.context['@id']);
         query.replaceKeyValue('assembly', defaultAssemblyByOrganism[tab]);
-        const newHref = query.format();
-        setdownloadHref(newHref);
+        context.navigate(query.format());
     };
 
     // Annotation type facet is not a normal facet, at least one annotation must be selected
     // Facet filters look like radiobuttons
     const chooseAnnotationType = (type) => {
-        let newAnnotationType = annotationType;
-        if (annotationType.indexOf(type) > -1 && annotationType.length > 1) {
-            newAnnotationType = annotationType.filter((annotation) => annotation !== type);
-        } else if (annotationType.indexOf(type) === -1) {
-            newAnnotationType = [...annotationType, type];
+        const query = new QueryString(props.context['@id']);
+        const existingAnnotations = query.getKeyValues('annotation_type');
+        if (existingAnnotations.indexOf(type) > -1 && existingAnnotations.length > 1) {
+            query.deleteKeyValue('annotation_type', type);
+        } else if (existingAnnotations.indexOf(type) === -1) {
+            query.addKeyValue('annotation_type', type);
+        } else {
+            query.replaceKeyValue('annotation_type', 'candidate Cis-Regulatory Elements');
         }
-        setAnnotationType(newAnnotationType);
+        context.navigate(query.format());
+    };
+
+    const handleReset = () => {
```

Comment:
The UI reset button is not intuitive. It took me time to figure out what it does. Maybe a tooltip (placeholder) would help. Or maybe set it like other facets on other pages are set (a link called "clear filters")

Reply:
Hm, I'm not sure what to do. I think it doesn't make sense to call it clear filters because it doesn't clear them all, it resets them to their default values.